modules/xmpp/JingleSessionPC.js

```diff
@@ -290,18 +324,32 @@ export default class JingleSessionPC extends JingleSession {
                 this,
                 this.peerconnection.iceConnectionState);
             switch (this.peerconnection.iceConnectionState) {
+            case 'checking':
+                this._iceCheckingStartedTimestamp = now;
+                break;
             case 'connected':
                 // Informs interested parties that the connection has been
                 // restored.
                 if (this.peerconnection.signalingState === 'stable') {
                     if (this.isreconnect) {
                         this.room.eventEmitter.emit(
                             XMPPEvents.CONNECTION_RESTORED, this);
-                    } else if (!this.wasConnected) {
-                        this.room.eventEmitter.emit(
-                            XMPPEvents.CONNECTION_ESTABLISHED, this);
                     }
+                }
+
+                if (!this.wasConnected && this.wasstable) {
+                    let eventName = this.isP2P ? 'p2p.ice.' : 'ice.';
+
+                    eventName += this.isInitiator ? 'initiator.' : 'responder.';
+                    eventName += 'checksDuration';
+                    Statistics.analytics.sendEvent(
+                        eventName,
```

Comment:
Probably isn't worth it, but how about we also put the total connection connection time? In theory we should have AVG(connectionEstablishmentDuration) = AVG(gatheringDuration) + AVG(checkingDuration), so this would serve only as a check.